### PR TITLE
APIContract/authentication-userProfile

### PR DESCRIPTION
**Pull request by ssanabia2**
Features: guest account, password recovery and change, view stats of a user's profile, change profile picture and new schemas

### CODE REVIEW

**Review by diegolopezl — APPROVED** [review]

File: api-contract/contract.yaml [hunk 1]
```diff
@@ -25,7 +27,25 @@ paths:
...
+      - authentication
+      summary: Create guest account
+      description: This can be done without being authenticated.
+      operationId: createGuest
+      requestBody:
+        description: Created guest object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Guest'
```
Comment (kajahno) [inline api-contract/contract.yaml:48]: In this one I don't think there's need to create a guest account at all. The idea is that the /user/{username} GET call and be called both authenticated and not-authenticated, so there is this double functionality in the same API

File: api-contract/contract.yaml [hunk 2]
```diff
@@ -181,18 +218,76 @@ components:
...
+          type: string
+        email:
+          type: string
+        bio:
+          type: string
+        website:
+          type: string
+        location:
+          type: string
+        picture:
```
Comment (kajahno) [inline api-contract/contract.yaml:266]: I would have used something like `imageProfile`

File: api-contract/contract.yaml [hunk 3]
```diff
@@ -181,18 +218,76 @@ components:
...
         createdAt:
           type: string
         lastSeen:
           type: string
+        friends:
+          type: integer
+        likes:
+          type: integer
+        posts:
+          type: integer
```
Comment (kajahno) [inline api-contract/contract.yaml:278]: These three can be calculated, no need to add as part of the user. For example I would expect that an object called Friend would have a field userID, similar to likes and posts.

File: api-contract/contract.yaml [hunk 4]
```diff
@@ -181,18 +218,76 @@ components:
...
           type: string
         message:
           type: string
+
   securitySchemes:
     bearer_auth:
       type: http
       scheme: bearer
+
+
```
Comment (kajahno) [inline api-contract/contract.yaml:292]: No need for two extra spaces here

**Review by kajahno — COMMENTED** [review]
Thanks for these changes, they looks pretty much on point, so good job! I've left a few comments

**Pull request merged**